### DISCUSSION TITLE
Field validation ignored inside App

### DISCUSSION THREAD

**Discussion by minyan-gu** [directus/directus#10483] [Archive]
The field validation seems ignored even on `Number` type.

Tried:
1. Set the Min and Max values on the `interface` tab, doesn't work when create/update the item.
2. Set a Between rule inside the `condition` tab, doesn't work neither.

![image](https://user-images.githubuserconte
![image](https://user-images.githubusercontent.com/49973259/145852360-ccab835c-78e0-4d31-aa01-4a2fb00deabc.png)
nt.com/49973259/145851725-c97f3d97-c30f-4946-bb8b-70319e07f2f2.png)
![image](https://user-images.githubusercontent.com/49973259/145851759-92789edc-f86d-4931-85c8-39e9c8534061.png)
![image](https://user-images.githubusercontent.com/49973259/145852360-ccab835c-78e0-4d31-aa01-4a2fb00deabc.png)

### To Reproduce

1. Create a test collection with a single Number typed field.

2. Set the Min and Max values on the `interface` tab. e.g. Min: `0` and Max: `100`
3. Create a new item or edit existing one, input an out of range value still can save successfully. Expected behaviour: Show validation error when click save.

4. Also pls try to set a **_Between_** rule inside the `condition` tab, it doesn't work as well as nothing prevents the save action.

**Comment by rijkvanzanten** [comment]
Heya! I think this is mostly a slight misunderstanding in what it's supposed to do. The min / max value options of the number input simply set the "min" and "max" attributes of the underlying `<input>` element, which isn't the same as configuring the validation rules for the role.

Similarly, the conditions tab is used to allow you to configure interface option changes on a value of a field. This will allow you to show / hide a field based on the value of another field. Same as with the interface options, this is not data (/ API) validation. 

The validation rules you're looking for can be configured under _Permissions_ of the roles detail page 👍🏻

**Reply by minyan-gu** [reply]
Not sure if this solve all issues.

1. What's stopping us to have the JS validation based on input `min` and `max` attribute if the input type is `number` already.
2. If using your suggested approach, then the question would be how to enable those validations also to `Admin` users, and what's the way to share the same validation rule across multiple or even all roles?

For rules in condition tab, understand if the use case is only for show/hide the fields, so thanks for provided clarity here.

**Reply by rijkvanzanten** [reply]
> What's stopping us to have the JS validation based on input min and max attribute if the input type is number already.

Nothing, it's simply not implemented like that at this point in time.

> If using your suggested approach, then the question would be how to enable those validations also to Admin users, and what's the way to share the same validation rule across multiple or even all roles?

This is tracked in https://github.com/directus/directus/discussions/7476 and https://github.com/directus/directus/discussions/4390
